torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -27,15 +27,32 @@ class ShardingFilterIterDataPipe(IterDataPipe):
     """
     def __init__(self, source_datapipe: IterDataPipe):
         self.source_datapipe = source_datapipe
+        self.sharding_groups: Dict[int, Any] = {}
```

Comment:
Should the type be `Dict[Optional[int], Tuple[int, int]]`? Because:
1. key can be `None`
2. value should always be `Tuple[int, int]`?

What makes you put `Any` instead?